tlscanary/modes/performance.py

```diff
@@ -60,80 +70,143 @@ def run(self):
+
+        # Tally overall performance of test vs release
+        self.total_change = self.calculate_speed_change(test_aggregate, base_aggregate)
+
+        for result in self.temp_hosts:
+            log.log(result)
 
         meta["run_finish_time"] = datetime.datetime.utcnow().isoformat()
-        meta["total_change"] = self.total_change
+        meta["performance_change"] = self.total_change
         self.save_profile(self.test_profile, "test_profile", log)
         self.save_profile(self.base_profile, "base_profile", log)
         log.stop(meta=meta)
 
+        # this will go away post-development
+        self.temp_debug_output()
+
+    # temporary debugging function
+    def temp_debug_output(self):
+        logging.info("Total change: %s" % self.total_change)
+        for i in xrange (0,3):
+            logging.info("Host: %s " % self.temp_hosts[i]["host"])
+            logging.info("Average change: %s " % self.temp_hosts[i]["connection_speed_change_average"])
+            logging.info("Median change: %s " % self.temp_hosts[i]["connection_speed_change_median"])
+            logging.info(self.temp_hosts[i]["connection_speeds"])
+            logging.info(self.temp_hosts[i]["connection_speeds_base"])
+
+    @staticmethod
+    def calculate_average_speed(sample_list):
+        num_samples = len(sample_list)
+        running_total = 0
+        for i in xrange(0, num_samples):
+            running_total += sample_list[i]
+        return float(running_total / num_samples)
+
+    @staticmethod
+    def calculate_median_speed(sample_list):
+        sample_list.sort()
+        num_samples = len(sample_list)
+        middle = num_samples/2
+        if num_samples % 2 == 1:
+            median = sample_list[middle]
+        else:
+            median = (sample_list[middle] + sample_list[middle-1]) / 2
+        return float(median)
+
     @staticmethod
-    def extract_connection_speed(uri_set):
-        new_set = []
-        for record in uri_set:
-            new_set.append([
-                record[0],  # URI rank
-                record[2].response.response_time - record[2].response.command_time  # connection speed
-            ])
-        return new_set
-
-    def consolidate_connection_speed_info(self, uri_set, connection_sets):
-        for record in uri_set:
-            temp_speeds = []
-            speed_aggregate = 0
-            for connection_set in connection_sets:
-                speed = [d for d in connection_set if d[0] == record[0]][0][1]
-                speed_aggregate += speed
-                temp_speeds.append(speed)
-            record[2].response.connection_speed_average = speed_aggregate / self.args.scans
-            record[2].response.connection_speed_samples = str(temp_speeds).strip('[]')
+    def calculate_speed_change(a, b):
+        change = (float(a - b) / b) * 100
+        format_to_two_decimals = float('%.2f' % change)
```

Comment:
What's the reason to throw away digits here?

My gut feeling is to generally leave steps like this for the view / data presentation layers to decide.